Content.Client/Lobby/LobbyUIController.cs

```diff
@@ -352,7 +352,15 @@ public JobPrototype GetPreferredJob(HumanoidCharacterProfile profile)
     {
         var highPriorityJob = profile.JobPriorities.FirstOrDefault(p => p.Value == JobPriority.High).Key;
         // ReSharper disable once NullCoalescingConditionIsAlwaysNotNullAccordingToAPIContract (what is resharper smoking?)
-        return _prototypeManager.Index<JobPrototype>(highPriorityJob.Id ?? SharedGameTicker.FallbackOverflowJob);
+        // Frontier: Proper fallback for missing prototypes
+        //return _prototypeManager.Index<JobPrototype>(highPriorityJob.Id ?? SharedGameTicker.FallbackOverflowJob);
+        if (highPriorityJob.Id is { } highPriorityJobId &&
+            _prototypeManager.TryIndex<JobPrototype>(highPriorityJobId, out var job))
```

Comment:
Just coalesce it into a string.
```suggestion
        if (_prototypeManager.TryIndex(highPriorityJob, out JobPrototype? job))
```

Reply:
Thanks, boss!